libs/ui/design-system/src/lib/input.tsx

```diff
@@ -1,28 +1,66 @@
-import type * as React from 'react'
-
-import { Input as InputPrimitive } from '@nmit-coursition/ui/primitives/input'
+import * as InputPrimitive from '@nmit-coursition/ui/primitives/input'
 import { Label } from '@nmit-coursition/ui/primitives/label'
 import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
+
+type OverrideProps<T, V> = V & Omit<T, keyof V>
 
-type InputType = React.ComponentProps<'input'>['type']
+const inputVariants = cva('', {
+  variants: {
+    variant: {
+      primary: '',
+      verified: '', // green borders
```

Comment:
**suggestion:** Those are not variants, but styles that we can target with `:user-invalid` and `:user-valid`. Same thing with `disabled`.

Reply:
I just removed them :)